homeassistant/components/fritz/camera.py

```diff
@@ -0,0 +1,114 @@
+"""FRITZ camera integration.
+
+Currently only used to display QR codes.
+"""
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+import logging
+
+from homeassistant.components.camera import Camera
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_UNKNOWN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import dt as dt_util, slugify
+
+from .common import FritzBoxBaseEntity, FritzBoxTools
+from .const import DEFAULT_GUEST_WIFI_QR_REFRESH_SEC, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add a entities from a config_entry."""
+    _LOGGER.debug("Setting up FRITZ!Box camera entities")
+    fritzbox_tools: FritzBoxTools = hass.data[DOMAIN][entry.entry_id]
+
+    entities = await hass.async_add_executor_job(
+        _setup_guest_wifi_qr, fritzbox_tools, entry.title
+    )
+
+    async_add_entities(entities, True)
+
+
+def _setup_guest_wifi_qr(
+    fritzbox_tools: FritzBoxTools, device_friendly_name: str
+) -> list[FritzGuestWifiQRCamera]:
+    """Set up guest wifi entity to display the qr code."""
+    _LOGGER.debug("Setting up qr code camera entity")
+    ssid = fritzbox_tools.fritz_guest_wifi.ssid
+    return [
+        FritzGuestWifiQRCamera(
+            fritzbox_tools,
+            device_friendly_name,
+            ssid,
+        )
+    ]
+
+
+class FritzGuestWifiQRCamera(FritzBoxBaseEntity, Camera):
+    """Implementation of the FritzBox guest wifi QR code camera entity."""
+
+    QR_CODE = "QR code"
+
+    def __init__(
+        self,
+        fritzbox_tools: FritzBoxTools,
+        device_friendly_name: str,
+        ssid: str,
+    ) -> None:
+        """Initialize the camera."""
+        Camera.__init__(self)
+
+        self._attr_name = f"{device_friendly_name} {ssid} {self.QR_CODE}"
+        self._attr_unique_id = (
+            f"{fritzbox_tools.unique_id}-{slugify(ssid)}-{slugify(self.QR_CODE)}"
```

Comment:
Is the ssid really stable?

https://developers.home-assistant.io/docs/entity_registry_index#unique-id

Reply:
What about something like:

```suggestion
            f"{fritzbox_tools.unique_id}-guest_qr_code"
```

Reply:
I've done that because the wifi switch entities are named with the ssid in name (https://github.com/home-assistant/core/blob/dev/homeassistant/components/fritz/switch.py#L276). If you change the ssid, these entities are not stable at all. The old switch becomes unavailable while there is an additional switch with the new ssid. Thought we should do it the same way here.

Reply:
I think it's fine to leave it as for the switch platform.
Let me know if any change is needed.

Simone

Reply:
We don't allow non stable IDs for unique_id.

Reply:
Fixed